README.md
# PathOfBuildingCN17173-Dev
 
这是S16赛季开始维护的国服POB项目，从原有项目上延续而来。

目前自动更新的功能还未完成，所以麻烦手动下载新版Release，手动覆盖文件进行更新。

如有Bug，请在Issue页面留下消息，也鼓励大家自己在代码中做出修改，提出PR。

## 脚本操作

生成manifest.xml:

    python update_manifest.py --in-place --set-version=${{ next-version }}

## 断点调试本项目的方法
1. vscode 安装 `EmmyLua` 这个插件
2. 放开注释 `Launch.lua` 20-25行，修改路径`C:/Users/echo/` 为你自己的用户名目录
3. vscode 点左边debug栏，切换到 `EmmyLua 断点调试` 这个配置
4. 启动 `Path of Building.exe`
5. vscode 里面按 F5

changelog.txt
S21.0.6:2023/3/9
bugfix: 圣宗意识圣殿，魔力消耗总降报错

S21.0.5:2023/02/22
feat:   Data目录下新增翻译数据，进一步翻译计算界面
bugfix: 头盔附魔狙击之印诅咒效果
        永恒珠宝对小天赋点的影响有问题
        分享装备后打开报错
        邪恶之眼解析
        拖拽技能调整顺序报错
        血肉奉献迷宫附魔解析

S21.0.4:2023/02/14
feat:   更新加群按钮
        兼容国服市集游戏里面复制的希望之线
        项目调试说明以及报错解决
        真相之担 生命再生解析
bugfix: 高等不屈先驱技能报错
        法术击中回血、击中诅咒解析
        修复生效期间降低魔力

S21.0.3:2023/02/08
感谢Liang在本赛季对国服POB代码更新的贡献！
我创建了一个QQ群318952371，欢迎大家进群交流和反馈。
feat:   可以进行永恒珠宝的搜索和生成了
        更新了S21的暗金、天赋和词缀
bugfix: 修复了大量词缀的解析和计算问题

S21.0.2:2023/01/21
fixbug: 
修复瓶中词条，以及解析，
修复药剂生效期间各种
修复茁育助长承伤计算

S21.0.2:2023/01/17
fixbug: 修复禁域面容
S21.0.2:2023/01/16
fixbug: 修复爆炸箭矢的计算
S21.0.2:2023/01/15
fixbug:
寒焰的文案以及解析错误
夜吟解析错误
摧残之光抗性计算错误
裂空者
生效期间抗性
S21.0.1:2023/01/04
feat:
毁灭的逼近
咒逐
永恒诅咒 正常解析了
茁育助长
女巫克星
熵能毁灭
理性主义
烈火之歌
尼米斯
暴风遮罩
奥瑞亚的尽头
黎明行者
命运之风
禁域面容
回声锻炉
威尔玛的报偿 正常解析了
德瑞之舌
永恒之力
S21.0.1:2022/12/30
feat: 
开膛斧
王者之刃
奇塔弗之盛宴
狮眼的战弓
魔暴之痕
嗜血之爪
精华收割器
比诺的厨刀
戒律之影
尔奇的巨灵之锤
黑鲨
塑星者

S21.0.1:2022/12/30
feat: 
费伯之牙
阿兹里的刑刃
卡斯普里意志
伊芙班的诡计
异教面纱

S20.0.6:2022/9/8
bugfix: 仿品西里憎恨保留解析
        导入能量之刃出错
        脆弱最大值描述
        属性专精解析,欺诈升华博学者解析
        青影梦魇数值错误
        崇高愿景禁用了所有光环
        尊严三相解析
        希望之线巨圈解析
        特殊球解析
        菁华蠕虫词缀不正确
        聚风和提速尾流未生效
        誓约高等施法回响解析
        塑泥锤词缀解析
        西里的真相精准等级
        召唤伤害套用自身
        完美姿态仿品数值未更新

S20.0.5:2022/8/30
feat:   配置界面大改，修复了很多相关bug
        更新内核，支持中文路径（感谢草樣年華）
bugfix: 将军之吼报错
        精神失常解析
        完美姿态解析
        幻影射手计算总DPS报错
        结界相关装备报错
        守望之眼词条不一致
        元素专精-随机元素伤害解析
        暴击率随法术伤害压制率提高解析
        战斗法师解析
        爆炸箭矢层数不生效

S20.0.4:2022/8/28
bugfix: 召唤魔侍在计算界面崩溃的问题
        穿刺默认最大层数为0的问题

S20.0.3:2022/8/28
bugfix: 添加不完善的计算导致的bug

S20.0.2:2022/8/28
feat:   更新永恒珠宝相关内容，现在可以根据编号自动解析数据了
        更新大部分效果的计算方式
bugfix: 修复导入的无所遁形无法解析的问题
        修复范围珠宝（无所遁形、超自然、永恒珠宝等）的解析问题
style:  更换字体
        更新部分翻译

S20.0.1:2022/8/27
feat:   修改复杂暗金的生成和解析方式，包括禁断珠宝、无所遁形、崇高愿景等
        更新部分翻译
bugfix: 群星之灰技能石品质解析
        召唤生物攻击速度导致报错
        外部导入的禁断珠宝无法生效

S20.0.0:2022/8/26
感谢笨笨猪对国服POB的大力支持！新赛季更新内容较多，后续会持续修复问题和增加新功能
feat:   更新S20天赋树、技能和词缀
        更新暗金词缀和解析
        由于之前使用的代码仓库无法访问，旧版本无法自动更新到新版本，需要重新下载。添加仓库更新功能，以后更换仓库不需要重新下载。
        更换了新的运行时感谢（草樣年華），更换了字体，显示效果应该提高了。
        支持添加灭界者和焚界者基底。
bugfix: 近期使用召唤技能配置无法出现
        修复法师之血无法使药剂自动生效的问题
        修复复合属性多次计算全知的问题

S18.0.2:2022/2/22
bugifx: 酷寒凝视头盔词缀错误
        漩涡附魔解析失败
        尝试解决无限自动更新问题

S18.0.1:2022/2/20
feat:   防御计算重构，更新文本描述和显示效果
        摧残之光、毁灭激波、禁断之火、禁断之肉、无尽盛宴、贪婪之潮、漆黑天顶、破晓者解析
        添加几个基石天赋的解析
        添加技能层数mod汉化脚本
        支持skillTypeList的写法
bugfix: 诅咒光环的保留效能解析
        灵巧效果和夜刃解析
        带层数的技能无法配置层数
        捷和光环计数问题
        捷技能保留无效
        召唤升华文本有误
        忆境戒指未更新词缀及解析
        错误的打印函数
        解析投射物速度也作用于弓伤害天赋
        计算界面数字类型错误

S18.0.0:2022/2/13
feat:   更新S18天赋树、技能和暗金

S17.1.11:2021/12/3
bugfix: 负电甲解析

S17.1.10:2021/12/3
bugfix: 骸骨奉献额外格挡率附魔解析
        灵能护体不能充能解析
        悖论解析
        在你的冰缓区域里的敌人身上的诅咒效果提高
        鸟衣更新
        周围敌人的抗性降低解析
        水火不容元素易伤解析
feat:   计算界面添加一行显示持续伤害的Multiplier数值
        龙卷射击和三位一体随机转元素解析
        更新防御计算

S17.1.9:2021/11/23
bugfix: 瑞佛诅咒解析
        威能法印配置不生效
feat:   新版元素之相解析

S17.1.8:2021/11/20
bugfix: 毁面者汉化更新
        线圈汉化更新
        暴徒升华献祭怒火增伤不生效
        狂乱之相解析
        灵能护体解析
feat:   珠宝槽UI汉化

S17.1.7:2021/11/13
bugfix: 人格分裂解析
        仿品霜息解析
        梅吉诺德手套解析

S17.1.6:2021/11/13
bugfix: 邪眼之冠报错

S17.1.5:2021/11/13
feat:   UI汉化更新
bugfix: 暴击辅助+3解析失败
        坚忍专注 → 忍耐专注
        仿品终息解析
        修正"附带伤害"词缀解析问题，词缀文本错误问题 (黑子鱼咖)
        修正"为生存而战"词缀解析问题 (黑子鱼咖)
        修正"人格分裂"词缀解析问题 (黑子鱼咖)
        修正"水火不容"词缀解析问题 (黑子鱼咖)

S17.1.4:2021/11/9
bugfix: 修正取消技能天赋点时，预览界面有 '7'字样的问题 (黑子鱼咖)
        药剂持续时间为0
        极地之眼更新
        新药剂珠宝汉化
        替换"巫灵斗士"为"蛊灵斗士" (黑子鱼咖)

S17.1.3:2021/11/6
feat:   处理角色列表导入成功但天赋树导入401的问题（官方真会整活）
        按钮输入框汉化优化
        超然飞升解析

S17.1.2:2021/11/6
feat:   元素异常效果和持续时间解析
bugfix: 异色鞋导入失败

S17.1.1:2021/11/4
feat:   天赋树翻译库更新
        界面汉化更新
        尊崇猛攻、法师克星、号召冷却专精、近期地雷爆伤解析
        导入角色功能更新（感谢踩蘑菇@玩家00145072提供思路）
bugfix: 新版巨铜之顶词缀异常
        法术躲避相关暗金未更新
        寒冬宝珠层数
        触发暗影印记词缀
        吞噬者王冠数值
        插入的技能石保留
        空明之掌解析
        逃亡女王数值
        调整ParserEn顺序
        深渊珠宝召唤电点伤
        追命死灵描述异常
        仿品突围者解析异常

S17.1.0:2021/11/3
feat:   天赋树的词缀解析前进行翻译，采用原版Parser，大部分新天赋都可以处理了（终于不用一条条处理中文解析了）
bugfix: 新版巨铜之顶词缀异常

S17.0.7:2021/11/3
feat:   更新角色信息接口401的处理。目前国服角色信息接口需要登录，手动添加sessionid后可以获得角色信息。

S17.0.6:2021/11/2
feat:   更新天赋解析
bugfix: 异色鞋、双玉戒指导入问题
        新的星团天赋解析问题

S17.0.5:2021/11/1
feat:   更新天赋解析
bugfix: 创建的星团珠宝无珠宝槽

S17.0.4:2021/10/31
feat:   更新赞赏链接
        添加新技能
        更新词缀汉化
        解析部分新天赋
星团问题：由于汉化更新，旧版POB转化以后星团珠宝的基底解析可能出错，需要重新导入或者重新创建珠宝

S17.0.3:2021/10/29
bugfix: 关于界面报错

S17.0.2:2021/10/29
feat:   更新保留效率解析
bugfix: 灵能护盾解析

S17.0.1:2021/10/29
feat:   更新S17天赋树、传奇装备

S16.1.5:2021/10/18
feat: 更新汉化，由手动更新改为自动从GGPK中解析

S16.1.4:2021/09/11
bugfix: 保留精确小数点一位
        逝日数值
        忆境基底法术双倍不生效
        持续伤害中星团解析
        贵族锐眼升华
        黯黑贤者

S16.1.3:2021/08/30
feat: 触发技能耗蓝
bugfix: 导入的星团基底无法解析
        战斗专注 赤红珠宝的描述
        直攻要害双倍不生效
        虚空制箭者
        药剂效果词缀，可能有问题
        头盔诅咒附魔

S16.1.2:2021/08/30
bugfix: 极地装甲报错
        新版战斗专注珠宝解析
        空明拳解析
        遗产瓶中、欧克、阿兹里

S16.1.1:2021/08/26
bugfix: 投射物中星团解析

S16.1.0:2021/08/26
bugfix: 大部分大星团、少部分中星团的天赋解析
        独行使者、苦难秘辛解析，其他暗金小星团还未完全修复
        鞋子附魔词缀解析
        遗产瓶中、欧克和阿兹里
        卫士挑战球解析
        副手技能无基础暴击率的问题
        召唤死神描述出错
        沧海桑田数值下调
feat:   增加怒火中烧
        恢复自动更新功能

S16.0.1:2021/08/22
bugfix: 由于Cache导致天赋没有正确解析

S16.0.0:2021/8/21
这是S16赛季开始维护的国服POB项目，从原有项目上延续而来。
目前自动更新的功能还未完成，所以麻烦手动下载新版Release，手动覆盖文件进行更新。
如有Bug，请在Issue页面留下消息，也鼓励大家自己在代码中做出修改，提出PR。

Data/Misc.lua
-- This file is automatically generated, do not edit!

local data = ...
-- From DefaultMonsterStats.dat
data.monsterEvasionTable = { 67, 86, 104, 124, 144, 166, 188, 211, 234, 259, 285, 311, 339, 368, 397, 428, 460, 493, 527, 563, 600, 638, 677, 718, 760, 804, 849, 896, 944, 994, 1046, 1100, 1155, 1212, 1271, 1332, 1395, 1460, 1528, 1597, 1669, 1743, 1819, 1898, 1979, 2063, 2150, 2239, 2331, 2426, 2524, 2626, 2730, 2837, 2948, 3063, 3180, 3302, 3427, 3556, 3689, 3826, 3967, 4112, 4262, 4416, 4575, 4739, 4907, 5081, 5260, 5444, 5633, 5828, 6029, 6235, 6448, 6667, 6892, 7124, 7362, 7608, 7860, 8120, 8388, 8663, 8946, 9237, 9536, 9844, 10160, 10486, 10821, 11165, 11519, 11883, 12258, 12643, 13038, 13445, }
data.monsterAccuracyTable = { 14, 15, 15, 16, 17, 18, 19, 20, 21, 23, 24, 25, 26, 28, 29, 31, 32, 34, 35, 37, 39, 41, 43, 45, 47, 49, 52, 54, 57, 59, 62, 65, 68, 71, 74, 77, 81, 84, 88, 92, 96, 100, 105, 109, 114, 119, 124, 129, 135, 140, 146, 152, 159, 165, 172, 179, 187, 195, 203, 211, 220, 229, 238, 247, 257, 268, 279, 290, 301, 314, 326, 339, 352, 366, 381, 396, 412, 428, 444, 462, 480, 499, 518, 538, 559, 580, 603, 626, 650, 675, 701, 728, 755, 784, 814, 845, 877, 910, 945, 980, }
data.monsterLifeTable = { 22, 26, 31, 36, 42, 48, 55, 62, 70, 78, 87, 97, 107, 119, 131, 144, 158, 173, 190, 207, 226, 246, 267, 290, 315, 341, 370, 400, 432, 467, 504, 543, 585, 630, 678, 730, 785, 843, 905, 972, 1042, 1118, 1198, 1284, 1375, 1472, 1575, 1685, 1802, 1927, 2059, 2200, 2350, 2509, 2678, 2858, 3050, 3253, 3469, 3698, 3942, 4201, 4476, 4768, 5078, 5407, 5756, 6127, 6520, 6937, 7380, 7850, 8348, 8876, 9436, 10030, 10660, 11328, 12036, 12787, 13582, 14425, 15319, 16265, 17268, 18331, 19457, 20649, 21913, 23250, 24667, 26168, 27756, 29438, 31220, 33105, 35101, 37214, 39450, 41817, }
data.monsterAllyLifeTable = { 15, 16, 18, 20, 22, 24, 27, 29, 32, 35, 38, 41, 44, 48, 52, 56, 60, 65, 70, 75, 81, 87, 93, 99, 106, 114, 121, 130, 138, 148, 158, 168, 179, 190, 203, 215, 229, 244, 259, 275, 292, 310, 328, 348, 369, 392, 415, 440, 465, 493, 522, 552, 584, 618, 653, 691, 730, 772, 816, 862, 910, 961, 1015, 1072, 1131, 1194, 1260, 1329, 1402, 1478, 1559, 1644, 1733, 1827, 1926, 2029, 2138, 2253, 2373, 2500, 2633, 2773, 2919, 3074, 3236, 3406, 3585, 3773, 3970, 4178, 4395, 4624, 4864, 5116, 5381, 5659, 5951, 6257, 6578, 6916, }
data.monsterDamageTable = { 4.9899997711182, 5.5599999427795, 6.1599998474121, 6.8099999427795, 7.5, 8.2299995422363, 9, 9.8199996948242, 10.699999809265, 11.619999885559, 12.60000038147, 13.640000343323, 14.739999771118, 15.909999847412, 17.139999389648, 18.450000762939, 19.829999923706, 21.290000915527, 22.840000152588, 24.469999313354, 26.190000534058, 28.010000228882, 29.940000534058, 31.959999084473, 34.110000610352, 36.360000610352, 38.75, 41.259998321533, 43.909999847412, 46.700000762939, 49.650001525879, 52.75, 56.009998321533, 59.450000762939, 63.080001831055, 66.889999389648, 70.910003662109, 75.129997253418, 79.580001831055, 84.26000213623, 89.180000305176, 94.349998474121, 99.800003051758, 105.51999664307, 111.5299987793, 117.86000061035, 124.5, 131.49000549316, 138.83000183105, 146.5299987793, 154.63000488281, 163.13999938965, 172.07000732422, 181.44999694824, 191.30000305176, 201.63000488281, 212.47999572754, 223.86999511719, 235.83000183105, 248.36999511719, 261.5299987793, 275.32998657227, 289.82000732422, 305.01000976562, 320.94000244141, 337.64999389648, 355.17999267578, 373.54998779297, 392.80999755859, 413.01000976562, 434.17999267578, 456.36999511719, 479.61999511719, 504, 529.53997802734, 556.29998779297, 584.34997558594, 613.72998046875, 644.5, 676.75, 710.52001953125, 745.89001464844, 782.94000244141, 821.72998046875, 862.35998535156, 904.90002441406, 949.44000244141, 996.07000732422, 1044.8900146484, 1096, 1149.5, 1205.5, 1264.1099853516, 1325.4499511719, 1389.6400146484, 1456.8199462891, 1527.1199951172, 1600.6800537109, 1677.6400146484, 1758.1700439453, }
data.monsterArmourTable = { 22, 26, 31, 36, 42, 48, 55, 62, 70, 78, 87, 97, 107, 119, 131, 144, 158, 173, 190, 207, 226, 246, 267, 290, 315, 341, 370, 400, 432, 467, 504, 543, 585, 630, 678, 730, 785, 843, 905, 972, 1042, 1118, 1198, 1284, 1375, 1472, 1575, 1685, 1802, 1927, 2059, 2200, 2350, 2509, 2678, 2858, 3050, 3253, 3469, 3698, 3942, 4201, 4476, 4768, 5078, 5407, 5756, 6127, 6520, 6937, 7380, 7850, 8348, 8876, 9436, 10030, 10660, 11328, 12036, 12787, 13582, 14425, 15319, 16265, 17268, 18331, 19457, 20649, 21913, 23250, 24667, 26168, 27756, 29438, 31220, 33105, 35101, 37214, 39450, 41817, }
-- From MonsterVarieties.dat combined with SkillTotemVariations.dat
data.totemLifeMult = { [1] = 1, [2] = 1, [3] = 1, [4] = 1.2, [5] = 1, [6] = 1.2, [7] = 1.2, [8] = 1.2, [9] = 1, [10] = 1, [11] = 1, [12] = 1, [13] = 1.2, [15] = 1.2, [16] = 7.44, [17] = 1.2, [18] = 1, [19] = 1, [20] = 1.2, }
data.monsterVarietyLifeMult = {
	["枯骨野兽"] = 2.38,
	["腐化兽"] = 2.38,
	["灭世眠兽"] = 2.38,
	["怒啄恐喙鸟"] = 0.7,
	["枯骨恐喙鸟"] = 1.12,
	["腐化的恐喙鸟"] = 1.12,
	["食腐骨喙"] = 0.99,
	["腐败恐喙"] = 1.12,
	["宿命弓手"] = 0.96,
	["远古骨弓手"] = 0.96,
	["永恒骨弓"] = 0.96,
	["永恒猎手"] = 0.96,
	["魔侍射手"] = 0.96,
	["易碎哨探"] = 0.96,
	["易碎骨弓"] = 0.96,
	["易碎骨人"] = 0.96,
	["盗猎者残骨"] = 0.96,
	["沙化骨弓手"] = 0.96,
	["沙地哨探"] = 0.96,
	["烈焰骨弓手"] = 0.96,
	["风化的守望者"] = 0.96,
	["风化守望者"] = 0.96,
	["宝库猎者"] = 0.96,
	["脆弱的骨弓"] = 0.96,
	["罪恶之弓"] = 0.96,
	["雷霆仆役"] = 0.96,
	["瘟疫弓手"] = 0.96,
	["嗜血枯骨"] = 0.96,
	["被遣的逝者"] = 0.96,
	["咒灵射手"] = 0.96,
	["骨覆射手"] = 0.96,
	["缠骨弓手"] = 0.96,
	["染血弓手"] = 0.96,
	["血痕弓手"] = 0.96,
	["阿兹莫里先人"] = 0.96,
	["骨弓手"] = 1.4,
	["冰霜狱卫"] = 0.84,
	["冰霜骨法"] = 0.84,
	["冰霜教士"] = 0.84,
	["冰霜骨牧"] = 0.84,
	["冰霜守护者"] = 0.84,
	["冰霜吟唱者"] = 0.84,
	["冰霜响尾蛇"] = 0.84,
	["冰霜先锋"] = 0.84,
	["玛拉凯的启蒙"] = 1.3,
	["玛拉凯的启迪"] = 1.3,
	["金色吟唱者"] = 0.84,
	["微光圣徒"] = 0.84,
	["漆黑吟唱者"] = 0.84,
	["渎灵贤者"] = 0.84,
	["枯骨之炎"] = 0.84,
	["火炎骨法"] = 0.84,
	["火炎教士"] = 0.84,
	["火炎骨牧"] = 0.84,
	["火炎先锋"] = 0.84,
	["炼狱魔导士"] = 0.84,
	["火炎响尾蛇"] = 0.84,
	["火焰吟唱者"] = 0.84,
	["雷电狱卫"] = 0.84,
	["卢库库的仆从"] = 0.48,
	["雷电骨法"] = 0.84,
	["雷电教士"] = 0.84,
	["雷电先锋"] = 0.84,
	["雷电骨牧"] = 0.84,
	["雷电守护者"] = 0.84,
	["雷电吟唱者"] = 0.84,
	["雷电战骸"] = 0.84,
	["雷电响尾蛇"] = 0.84,
	["雷能枯灵"] = 0.84,
	["远古遗骸"] = 0.9,
	["圣徒遗骨"] = 0.9,
	["盗贼残骨"] = 0.9,
	["逝者枯骨"] = 0.9,
	["宝库守护者"] = 0.9,
	["永恒典狱长"] = 0.9,
	["永恒流浪者"] = 0.9,
	["恶魔守望者"] = 0.9,
	["沙虫巫师"] = 0.9,
	["镰刀尸骨"] = 1.58,
	["灰烬收割者"] = 1.58,
	["愤怒骨灵"] = 0.9,
	["还魂圣人"] = 0.9,
	["复生圣徒"] = 0.9,
	["宝库卫士"] = 0.9,
	["沙之守卫"] = 0.9,
	["沙地守护者"] = 0.9,
	["远古工人"] = 0.9,
	["上古行者"] = 0.9,
	["角斗士残骨"] = 0.9,
	["易碎斗士"] = 0.9,
	["还魂角斗士"] = 1.38,
	["复生卫士"] = 1.38,
	["脆弱的土匪"] = 0.9,
	["脆弱的农夫"] = 0.9,
	["脆弱农夫"] = 0.9,
	["焚烬之民"] = 0.9,
	["梦魇之民"] = 0.9,
	["骨覆战士"] = 0.9,
	["缠骨武士"] = 0.9,
	["染血战士"] = 0.9,
	["血痕武士"] = 0.9,
	["远古囚徒"] = 0.9,
	["巨型骸骨斗士"] = 1.98,
	["觉醒巨人"] = 1.98,
	["冰霜祭骨"] = 1.08,
	["冰霜斗者"] = 1.08,
	["冰霜卫者"] = 1.08,
	["福克的冰霜卫士"] = 1.08,
	["火炎祭骨"] = 1.08,
	["火炎斗者"] = 1.08,
	["火炎卫者"] = 1.08,
	["福克烈焰卫士"] = 1.08,
	["费尔的船员"] = 1.68,
	["迷失的船员"] = 0.9,
	["海盗领主的船员"] = 1.68,
	["灵骸守卫"] = 1.08,
	["雷能枯骨"] = 1.08,
	["雷鸣守护者"] = 1.08,
	["雷电卫者"] = 1.08,
	["福克风暴卫士"] = 1.08,
	["薛朗的崇拜者"] = 1.08,
	["魔侍工程师"] = 0.9,
	["不死工程师"] = 0.9,
	["焚烬枯骨"] = 1.08,
	["冰霜枯骨"] = 1.08,
	["雷霆枯骨"] = 1.08,
	["碎裂喷炎者"] = 0.84,
	["碎裂唤霜者"] = 0.84,
	["碎裂风暴使者"] = 0.84,
	["稻草人"] = 0.7,
	["饥饿腐尸"] = 0.72,
	["路霸"] = 1.08,
	["浮灵"] = 1.08,
	["无思者"] = 1.08,
	["爆裂浮灵"] = 1.08,
	["浮灵拥护者"] = 1.08,
	["浸水腐尸"] = 1.08,
	["沉睡腐尸"] = 1.08,
	["禁忌祭者跟随者"] = 1.8,
	["禁忌祭者随从"] = 1.62,
	["禁忌祭者奴隶"] = 1.62,
	["禁忌祭者苏醒"] = 2.52,
	["苍白腐尸"] = 0.99,
	["噬灵腐尸"] = 1.08,
	["虔诚逝者"] = 1.08,
	["缠丝腐尸"] = 1.08,
	["咒怨腐尸"] = 1.08,
	["旱皮腐尸"] = 1.08,
	["沥青跛尸"] = 1.08,
	["脓血腐尸"] = 1.08,
	["贵族遗体"] = 1.08,
	["裹缚贵族"] = 1.32,
	["马拉克斯干尸"] = 1.08,
	["腐化角斗士"] = 1.08,
	["腐化老练角斗士"] = 1.08,
	["悲语骸骨"] = 0.9,
	["复生苦工"] = 0.9,
	["不灭之弦"] = 0.96,
	["不息弓手"] = 0.96,
	["悲语之弦"] = 0.96,
	["炽炎枯骨"] = 1.08,
	["炽炎术者"] = 0.84,
	["狂炎术者"] = 0.84,
	["焚尽法师"] = 0.84,
	["灵迹教士"] = 0.84,
	["苍白斗士"] = 1.08,
	["荒骨信徒"] = 1.08,
	["苍白教士"] = 0.84,
	["荒骨教士"] = 0.84,
	["搬运工遗骨"] = 0.96,
	["海盗遗骨"] = 0.9,
	["阿兹里的爱将"] = 2.21,
	["冥喙"] = 4,
	["玫瑰之影"] = 16,
	["棘刺之影"] = 16,
	["渎灵遗骸"] = 0.9,
	["冰雪祭者"] = 0.84,
	["风暴祭者"] = 0.84,
	["塑骨者欧西卡提"] = 2.2,
	["腐化的烬织者"] = 0.84,
	["腐化的霜织者"] = 0.84,
	["腐化遗骸"] = 0.9,
	["腐化阴尸"] = 1.08,
	["腐化的骸骨掠夺者"] = 0.9,
	["腐化的灼炎之矢"] = 0.96,
	["苍白的乌旗守卫"] = 1.35,
	["月影奴隶"] = 1.35,
	["苍白的指挥官"] = 1.95,
	["福尔堕灵"] = 1.3,
	["福尔的坠落"] = 1.3,
	["永恒魔侍"] = 1.3,
	["堕落之灵"] = 1.3,
	["邪秽之骨"] = 1.3,
	["不洁士兵"] = 1.3,
	["紫晶堕灵"] = 1.3,
	["阳洁堕灵"] = 1.3,
	["红痕堕灵"] = 1.3,
	["红之堕灵"] = 1.95,
	["不洁射手"] = 1.1,
	["红痕射手"] = 1.1,
	["福尔的先锋"] = 1.1,
	["紫晶侦察者"] = 1.1,
	["阳洁先锋"] = 1.1,
	["红痕先锋"] = 1.1,
	["红之先锋"] = 1.5,
	["腐化精英"] = 1.2,
	["充能工程师"] = 1.4,
	["冰冻建筑师"] = 1.4,
	["女神守护者"] = 1.2,
	["煤渣告发者"] = 1.08,
	["烧毁告发者"] = 1.08,
	["寒冰告发者"] = 1.08,
	["霜寒告发者"] = 1.08,
	["雷霆告发者"] = 1.08,
	["风暴告发者"] = 1.08,
	["苦寒诈欺者"] = 0.84,
	["沸腾诈欺者"] = 0.84,
	["放电诈欺者"] = 0.84,
	["古代挑战者"] = 0.9,
	["腐败挑战者"] = 1.08,
	["腐烂告发者"] = 1.08,
	["腐烂挑战者"] = 1.08,
	["古代斗士"] = 0.9,
	["迷宫碾压者"] = 1.98,
	["不死守卫"] = 0.9,
	["贵族魔卫"] = 1.3,
	["实验之体"] = 1.3,
	["实验魔卫"] = 1.3,
	["惊惧之轮"] = 1.28,
	["骨壳"] = 2.07,
	["白骨之牙"] = 2.07,
	["暴躁的矿工"] = 1.32,
	["不安的奴隶"] = 1.56,
	["幽暗蹒跚者"] = 1.08,
	["幽暗结核怪"] = 1.08,
	["尸浆亡者"] = 1.08,
	["复生的弄臣"] = 1.65,
	["无皮先锋"] = 1.5,
	["无皮哨探"] = 1.2,
	["忘却的骨灵"] = 1,
	["福尔的复仇"] = 1.1,
	["撕裂者"] = 3,
	["原始守望者"] = 1.7,
	["菌潮猎骨者"] = 6,
	["菌潮远古骨弓手"] = 4.8,
	["菌潮巨像"] = 13.5,
	["脆骨新兵"] = 1.35,
	["掘出的骨剑士"] = 1.82,
	["掘出的骨武士"] = 1.98,
	["掘出的骨弓手"] = 1.54,
	["掘出的莽夫"] = 2.4,
	["掘出的顽童"] = 1.74,
	["[DNT] Iron Point v2"] = 1.98,
	["[DNT] Chatter v2"] = 2,
	["掘出的灵枢"] = 2,
	["熊心盗贼"] = 0.9,
	["豪猪盗贼"] = 0.9,
	["鹿裘盗贼"] = 0.9,
	["洞穴兽"] = 2.38,
	["浴火不死塑像"] = 4,
}
-- From MonsterMapDifficulty.dat
data.mapLevelLifeMult = { [66] = 1.01, [67] = 1.03, [68] = 1.05, [69] = 1.09, [70] = 1.13, [71] = 1.19, [72] = 1.24, [73] = 1.3, [74] = 1.36, [75] = 1.43, [76] = 1.49, [77] = 1.55, [78] = 1.61, [79] = 1.68, [80] = 1.77, [81] = 1.87, [82] = 1.98, [83] = 2.09, [84] = 2.17, [85] = 2.25, [86] = 2.34, [87] = 2.42, [88] = 2.5, [89] = 2.58, [90] = 2.66, }
-- From MonsterMapBossDifficulty.dat
data.mapLevelBossLifeMult = { [1] = 1.5, [2] = 1.52, [3] = 1.54, [4] = 1.56, [5] = 1.58, [6] = 1.6, [7] = 1.62, [8] = 1.64, [9] = 1.66, [10] = 1.68, [11] = 1.7, [12] = 1.72, [13] = 1.74, [14] = 1.76, [15] = 1.78, [16] = 1.8, [17] = 1.82, [18] = 1.84, [19] = 1.86, [20] = 1.88, [21] = 1.9, [22] = 1.92, [23] = 1.94, [24] = 1.96, [25] = 1.98, [26] = 2, [27] = 2.02, [28] = 2.04, [29] = 2.06, [30] = 2.08, [31] = 2.1, [32] = 2.12, [33] = 2.14, [34] = 2.16, [35] = 2.18, [36] = 2.2, [37] = 2.22, [38] = 2.24, [39] = 2.26, [40] = 2.28, [41] = 2.3, [42] = 2.32, [43] = 2.34, [44] = 2.36, [45] = 2.38, [46] = 2.4, [47] = 2.42, [48] = 2.44, [49] = 2.46, [50] = 2.48, [51] = 2.5, [52] = 2.52, [53] = 2.54, [54] = 2.56, [55] = 2.58, [56] = 2.6, [57] = 2.62, [58] = 2.64, [59] = 2.66, [60] = 2.68, [61] = 2.7, [62] = 2.72, [63] = 2.74, [64] = 2.76, [65] = 2.78, [66] = 2.8, [67] = 2.8, [68] = 2.8, [69] = 3.03, [70] = 3.16, [71] = 3.32, [72] = 3.48, [73] = 5.63, [74] = 5.97, [75] = 6.32, [76] = 6.66, [77] = 6.99, [78] = 10.46, [79] = 11.32, [80] = 13.01, [81] = 15.43, [82] = 17.2, [83] = 19.64, [84] = 21.46, [85] = 21.46, [86] = 21.46, [87] = 21.46, [88] = 21.46, [89] = 21.46, [90] = 21.46, }
data.mapLevelBossAilmentMult = { [1] = 0.81, [2] = 0.81, [3] = 0.81, [4] = 0.81, [5] = 0.81, [6] = 0.81, [7] = 0.81, [8] = 0.81, [9] = 0.81, [10] = 0.81, [11] = 0.81, [12] = 0.81, [13] = 0.81, [14] = 0.81, [15] = 0.81, [16] = 0.81, [17] = 0.81, [18] = 0.81, [19] = 0.81, [20] = 0.81, [21] = 0.81, [22] = 0.81, [23] = 0.81, [24] = 0.81, [25] = 0.81, [26] = 0.81, [27] = 0.81, [28] = 0.81, [29] = 0.81, [30] = 0.81, [31] = 0.81, [32] = 0.81, [33] = 0.81, [34] = 0.81, [35] = 0.81, [36] = 0.81, [37] = 0.81, [38] = 0.81, [39] = 0.81, [40] = 0.81, [41] = 0.81, [42] = 0.81, [43] = 0.81, [44] = 0.81, [45] = 0.81, [46] = 0.81, [47] = 0.81, [48] = 0.81, [49] = 0.81, [50] = 0.81, [51] = 0.8, [52] = 0.79, [53] = 0.78, [54] = 0.77, [55] = 0.76, [56] = 0.75, [57] = 0.74, [58] = 0.73, [59] = 0.72, [60] = 0.71, [61] = 0.7, [62] = 0.69, [63] = 0.68, [64] = 0.67, [65] = 0.66, [66] = 0.65, [67] = 0.64, [68] = 0.63, [69] = 0.59, [70] = 0.58, [71] = 0.57, [72] = 0.56, [73] = 0.55, [74] = 0.53, [75] = 0.52, [76] = 0.51, [77] = 0.5, [78] = 0.49, [79] = 0.48, [80] = 0.48, [81] = 0.47, [82] = 0.47, [83] = 0.47, [84] = 0.47, [85] = 0.47, [86] = 0.47, [87] = 0.47, [88] = 0.47, [89] = 0.47, [90] = 0.47, }
